### PR TITLE
[Release Test] Send release test result to db pipeline

### PR DESCRIPTION
**Pull request by jjyao**
## Why are these changes needed?
- Send release test result to db pipeline
- Add perf metrics for microbenchmark so that we can alert on them

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/microbenchmark/run_microbenchmark.py [hunk 1]
```diff
@@ -19,6 +19,13 @@ def to_dict_key(key: str):
         f"{to_dict_key(v[0])}": (v[1], v[2]) for v in results if v is not None
     }
 
+    perf_metrics = [
+        {"perf_metric_name": to_dict_key(v[0]),
+         "perf_metric_value": v[1],
+         "perf_metric_type": "THROUGHPUT"} for v in results if v is not None
+    ]
+    result_dict["perf_metrics"] = perf_metrics
```
Comment (jjyao) [inline release/microbenchmark/run_microbenchmark.py:27]: For any tests that want perf alert, they should define perf_metrics like this.

**Comment by jjyao** [comment]
Tested the pipeline, it's working: https://buildkite.com/ray-project/periodic-ci/builds/2919

**Review by simon-mo — APPROVED** [review]

File: release/e2e.py [hunk 2]
```diff
@@ -792,6 +792,31 @@ def report_result(
...
+        "cluster_url": results["_session_url"],
+        "wheel_url": results["_commit_url"],
+        "runtime": results["_runtime"],
+        "stable": results["_stable"],
+    }
+
+    logger.debug(f"Result json: {json.dumps(result_json)}")
+
+    firehose_client = boto3.client("firehose", region_name="us-west-2")
+    firehose_client.put_record(
```
Comment (rkooo567) [inline release/e2e.py:814]: Do we need exponential retry here? (like S3 APIs we use)

**Review by rkooo567 — APPROVED** [review]

File: release/e2e.py (hunk 2 above)
Comment (krfricke) [reply release/e2e.py:814]: +1, or at least catch errors here

**Review by krfricke — APPROVED** [review]

**Comment by rkooo567** [comment]
oops. I saw this wasn't handled yet https://github.com/ray-project/ray/pull/22667#discussion_r816330506

@jjyao if it makes sense, can you create a follow up PR?

**Pull request merged**